src/services/i18n/I18n.ts

```diff
@@ -0,0 +1,40 @@
+import { createI18n } from '@kong-ui-public/i18n'
+
+import { camelCaseToWords } from '@/utilities/camelCaseToWords'
+import { get } from '@/utilities/get'
+
+interface I18nRecord {
+  [key: string]: I18nRecord | string
+}
+
+class I18nError extends Error {
+  get key() {
+    return this.message
+  }
+}
+
+export default <T extends I18nRecord>(strs: T) => {
+  const i18n = createI18n<typeof strs>('en-us', strs, true)
+  return {
+    ...i18n,
+    t: function (...rest: Parameters<typeof i18n['t']>) {
+      const key = rest[0]
+      try {
+        if (typeof get(strs, key) === 'undefined') {
+          if (key.startsWith('http.api.')) {
+            throw new I18nError(key)
```

Comment:
**Question**: Why is this handled via exception? To me, this looks like it could just be `return $fallbackMechanism`, no?

Reply:
I expected `core/i18n` to throw an error when a key wasn't found. It does throw an error but then it swallows that error and deals with it itself by just erroring to console, so we can't override that. Ideally via bubbling errors so thats how I continued to write it and I still prefer this.

I think apart from that we are probably going to be getting rid of this pretty soon anyway and just using `core/i18n` erroring and all our term > human mappings will be pre-built. This was just to support what we currently have until we do that.